ts/Iterable.ts

```diff
@@ -426,57 +439,70 @@ class Iterable {
   }
 
   // PRIVATE
-  private static setupEventHandlers(config: IterableConfig) {
-    if (config.urlHandler) {
+  private static setupEventHandlers() {
+     //Remove all listeners to avoid duplicate listeners
+     RNEventEmitter.removeAllListeners(EventName.handleUrlCalled)
+     RNEventEmitter.removeAllListeners(EventName.handleInAppCalled)
+     RNEventEmitter.removeAllListeners(EventName.handleCustomActionCalled)
+     RNEventEmitter.removeAllListeners(EventName.handleAuthCalled)
+
+    if (Iterable.savedConfig.urlHandler) {
       RNEventEmitter.addListener(
         EventName.handleUrlCalled,
         (dict) => {
           const url = dict["url"]
           const context = IterableActionContext.fromDict(dict["context"])
-          if (config.urlHandler!(url, context) == false) {
-            Linking.canOpenURL(url)
-              .then(canOpen => {
-                if (canOpen) { Linking.openURL(url) }
-              })
-              .catch(reason => { console.log("could not open url: " + reason) })
+          if (Platform.OS === "android") {
```

Comment:
I would move this condition inside the `wakeApp` method. Because there is no iOS implementation, it will probably crash if called on iOS. The only thing preventing this is this condition here, and it is within a lot of other logic, so it can be easy. Also, someone else can add a call to `wakeApp` in some other method down the road, and that would also require a condition on `Platform.OS`. It is much easier to just have the check within the `wakeApp` method.